src/runtime/runtime_rpi3.go

```diff
@@ -0,0 +1,89 @@
+// +build rpi3
+
+package runtime
+
+import (
+	"machine"
+	"unsafe"
+)
+
+const tickMicros = int64(1)
+
+type timeUnit int64
+
+//go:export sleepticks sleepticks
+func sleepTicks(n timeUnit) {
+
+	machine.WaitMuSec(uint64(n))
+}
+
+func ticks() timeUnit {
+	return timeUnit(machine.SystemTime())
+}
+
+//go:export main
+func main() {
+	run()
+	machine.Abort()
+}
+
+func putchar(c byte) {
+	machine.MiniUART.WriteByte(c)
+}
+
+
+// abort is called by panic().
+func abort() {
+	machine.Abort()
+}
+
+
+// called once the memory allocator is ready and heap initialized
+func postinit() {
+	// Initialize .bss: zero-initialized global variables.
```

Comment:
This doesn't make sense. You can't initialize the heap without first clearing the `.bss` section. You need to clear the BSS section (and initialize the `.data` section) before running any normal code.

Reply:
Fixed, see other comment.

Reply:
I still don't see where .bss is initialized? You have a `postinit` function which appears to do that, but it isn't called anywhere.
My proposal would be:

* Rename `postinit` to `preinit`.
* Call `preinit()` just before `run()` call.
* Add a new empty `postinit` function. It is only really used to enable interrupts on AVR but because how things are designed it has to be defined in every target.

Also I don't see where `.data` is initialized. It might not be necessary on the Raspberry Pi 3, though.